<launch>
<param name="use_sim_time" value="false"/>
<param   name="robot_description" textfile="$(find xtark_description)/urdf/xtark_description.urdf" />

<arg name="use_ekf" default="true"/>

<arg name="publish_odom_transform" default="true"/>

<node
    name="robot_state_publisher"
    pkg="robot_state_publisher"
    type="robot_state_publisher" />


<node name="xtark_driver" pkg="xtark_driver" type="xtark_driver" output="screen" respawn="false">
        <rosparam file="$(find xtark_driver)/config/xtark_config.yaml" command="load"/>
	      <param name="publish_odom_transform" value="false" if="$(arg use_ekf)"/>
	      <param name="publish_odom_transform" value="$(arg publish_odom_transform)" unless="$(arg use_ekf)"/>
        <remap from="/odom" to="/odom_raw" if="$(arg use_ekf)"/>
        <remap from="/odom" to="/odom" unless="$(arg use_ekf)"/>
</node>

<group if="$(arg use_ekf)">
    <node pkg="robot_pose_ekf" type="robot_pose_ekf" name="robot_pose_ekf" output="screen">
          <param name="output_frame" value="odom"/>
          <param name="base_footprint_frame" value="base_link"/>              
          <param name="freq" value="50.0"/>
          <param name="sensor_timeout" value="0.5"/>
          <param name="odom_used" value="true"/>
          <param name="imu_used"  value="true"/>
          <param name="vo_used"   value="false"/>
          <remap from="odom" to="/odom_raw"/>
          <remap from="/imu_data" to="/imu_raw"/>
          <remap from="/robot_pose_ekf/odom_combined" to="/odom_combined"/>
    </node>
    <node pkg="xtark_driver" type="odom_ekf.py" name="odom_ekf_node" output="screen">
          <remap from="input" to="/odom_combined"/>
          <remap from="output" to="/odom"/>
    </node>
</group>

</launch>
